DOCS-1570 - MCP Server GA - Docs (#6904)
SumoLogic/sumologic-documentation: docs/api/mcp-server.md

Author: not supported in MCP - follow edits [here](https://github.com/SumoLogic/sumologic-documentation/pull/6929/changes)

---
id: mcp-server
title: Sumo Logic MCP Server
sidebar_label: MCP Server ✨
description: Connect your AI tools to Sumo Logic via MCP to query logs, manage insights, and investigate security incidents using Claude Code CLI.
---

import useBaseUrl from '@docusaurus/useBaseUrl';

The Sumo Logic MCP server lets MCP clients (external AI models) connect to Sumo Logic to query logs, investigate security insights, manage alerts and dashboards, and more. Use natural language to bring Sumo Logic search, evidence, and platform context into the AI tools you already use, such as developer IDEs, security workflows, and enterprise AI platforms.

:::note
Prefer a built-in conversational experience instead of connecting an external AI client? The MCP server exposes discrete tools that any MCP-compatible client can call (optionally guided by the [sumo-investigator skill](#improve-investigations-with-a-skill)), while [Mobot](/docs/search/mobot) is Sumo Logic's own set of specialized agents — including the SOC Analyst Agent, Query Agent, Knowledge Agent, and monitor creation — built directly into the product.
:::

> <!-- Training/course link? -->   

Author: ```suggestion

import Iframe from 'react-iframe';

<!--
:::training Micro Lesson

Watch this micro lesson to learn how to connect an MCP-compatible AI client to Sumo Logic.

<Iframe url="https://fast.wistia.net/embed/iframe/ej2qcjc6l7?web_component=true&seo=true&videoFoam=false"
  width="854px"
  height="480px"
  title="Micro Lesson: MCP Server"
  id="wistiaVideo"
  className="video-container"
  display="initial"
  position="relative"
  allow="autoplay; fullscreen"
  allowfullscreen
/>

:::

-->
```

## Enable or disable the MCP server

MCP server access is enabled by default. An administrator can turn it on or off for your entire organization.

1. In the main Sumo Logic menu, select **Administration** > **Feature Management**.
1. In the **MCP Server access** row, use the **Enabled** toggle to turn the MCP server on or off.<br/><img src={useBaseUrl('img/api/mcp/mcp-feature-management.png')} alt="Feature Management page showing the AI features and MCP Server access toggles" style={{border: '1px solid gray'}} width="800" />

Enabling MCP Server access makes the server available for connection. Clients still authenticate with OAuth 2.0, and CIMD is enabled separately on the Policies page. See [Prerequisites](#prerequisites) and [OAuth Client Setup](/docs/manage/security/oauth).

Disabling the MCP server prevents MCP clients from connecting, but does not delete any data. MCP Server access is a separate setting from the **AI features** toggle, which governs Mobot, Parse Assist, and the SOC Analyst Agent, so you can enable or disable the MCP server independently of those capabilities.

## Configure in Claude Code CLI

> Claude Code CLI uses OAuth 2.0 with CIMD, so you do not need to create OAuth credentials before setup. Browser-based login handles authentication and token refresh automatically.

Author: ```suggestion
Claude Code CLI uses OAuth 2.0 with CIMD. You do not need to create OAuth credentials before setup. Browser-based login handles authentication and token refresh automatically.
```

   [CIMD](https://datatracker.ietf.org/doc/draft-ietf-oauth-client-id-metadata-document/) is the recommended authentication mechanism for MCP clients. You can learn more about how CIMD works at [client.dev](https://client.dev/). If you have any questions about client compatibility, contact [Sumo Logic Support](https://support.sumologic.com/support/s).
   :::

<!-- Training/course link? -->   

<!-- TODO (DOCS-1570): Add an "Enable or disable the MCP server" section once details are confirmed. Needs:
  - Exact UI path for the on/off control, and whether it's admin (org-wide) or per-user.
  - Default state at GA (on by default for AI-enabled orgs, or an admin must enable it).
  - AI opt-out behavior: whether a separate org setting disables the MCP server, where it lives, and whether it's the same opt-out that governs Mobot and the other agents.
  - Required permission/role to change it.

>   Source: the Dojo AI launch decks describe a self-service on/off toggle and note the MCP server is disabled for orgs that opted out of AI, but without UI specifics. -->

Author: @ganano @kdwink I'd like to a dd an "Enable or disable the MCP server" section. Could you help with the following?

1. Where is the on/off control? Exact UI path (e.g., Administration > … > something), and is it admin-level (org-wide) or per-user?
2. Default state at GA — is the MCP server on by default for AI-enabled orgs, or does an admin have to turn it on?
3. AI opt-out — is that a separate org setting (and where?) that, when enabled, disables the MCP server? Is it the same opt-out that governs Mobot/other agents, or MCP-specific?
4. Any permission/role required to change it.

Reviewer: Enabled by default. Its on the new Feature Management page. See the attached screenshots.

<img width="2169" height="1362" alt="Image" src="https://github.com/user-attachments/assets/78cd1c00-65ab-4084-a017-05058d4e6374" />

<img width="2160" height="1399" alt="Image" src="https://github.com/user-attachments/assets/cf532bd6-af7b-4952-bf5b-ead1b8c07e78" />

Author: thank you!

1. Select **Authenticate** and log in to the new org.

:::note

> If you previously granted consent for an org, you will not be prompted again. To revoke consent, go to your Sumo Logic user settings, then click **Personal Authorized Apps** and remove the app.

Author: ```suggestion
If you previously granted consent for an org, you will not be prompted again. To revoke consent, go to your Sumo Logic user settings and remove the app under **Personal Authorized Apps** (next to Personal Access Tokens).
```

Claude Code can invoke a skill in two ways:
* **Automatically**. Claude matches your question against the `description` and `when_to_use` context defined in the skill's frontmatter and invokes the skill without you needing to reference it directly.
* **Explicitly**. Type a slash command that matches the skill's folder name, for example, `/sumo-investigator`.

The skill linked below is a starting point based on Sumo Logic's internal testing of MCP tool workflows. It defines the agent's tool-use workflow (discover → sample → target) for log searches, plus guidance for triaging alerts, insights, and detection rules. Treat it as a foundation you can customize for your environment, or use as a model for additional skills of your own.

:::note
As the Sumo Logic MCP server evolves, for example, as tools are added, removed, or renamed, you may need to update this skill to match.
:::

1. Download <a href={useBaseUrl('files/sumo-investigator.zip')} target="_blank">sumo-investigator.zip</a> and unzip it.

> 1. Move the unzipped `sumo-investigator` folder into your skills directory: `.claude/skills/` for a project-specific skill available only in the current directory, or `~/.claude/skills/` to make it available across all projects. The result should be a `SKILL.md` file at `.claude/skills/sumo-investigator/SKILL.md` (or the equivalent path under `~/.claude/skills/`).

Author: @kdwink @ganano , after the user adds the `sumo-investigator` skill, do they need to go into Claude Desktop and click "Connect" on the environment(s) that show up?</br></br>

<img width="700" alt="Image" src="https://github.com/user-attachments/assets/887e7d0d-70aa-4daa-b543-f6d2a28aafe1" />

Reviewer: I don't know about Claude Desktop, I don't have a Claude Desktop account....at some point in their setup process they will need to connect, though I would suspect that you could add the skill either before or after connecting.

Author: Since this doc is scoped to Claude Code CLI (not Desktop, given the open Desktop connector issue), this is already handled order-agnostically — step 3 has users run `/mcp` to confirm the connection after adding the skill, regardless of when they connected. No content change needed here. Will revisit if/when we add Desktop-specific guidance.
